woocommerce/woocommerce
# Remove Active Shipping Zones check for displaying shipping calculator on Cart Page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:







We have an existing implementation that hides the shipping calculator button when the shipping address is not provided, no default customer location is set by the store, and the merchant has only one shipping zone that applies to all locations. In this scenario, requesting the shopper to enter an address is redundant, as it will not impact the shipping rates.

In #43803 PR, the `ActiveShippingZones` was introduced to check if there were any active fallback shipping zones, and [logic was added that exposed the shipping zones](https://href.li/?https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/BlockTypes/Cart.php#L248) to the front end.

With this PR, we are reverting the changes for Active Shipping Zones by eliminating the check for Active Shipping Zones to display the shipping calculator on the Cart Page. The shipping calculator will be visible if there are default shipping methods available.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Disable local pickup from wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location.  
2. Delete all shipping zones. 
3. Go to shipping settings at `wp-admin/admin.php?page=wc-settings&tab=shipping&section=options`.
4. Disable the "Hide shipping costs until an address is entered" option. 
5. Activate the "Enable the shipping calculator on the cart page" option.
![image](https://github.com/woocommerce/woocommerce/assets/11503784/6952c0d6-e7bb-4863-8ecb-1a9c64dcc3a7)

6. Verify "Default customer location" is "no location by default" at `wp-admin/admin.php?page=wc-settings`. 

<img width="1072" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/bf52ad51-373f-4f08-b838-1ca5d65214bf">

7. Navigate to the shipping zone and add shipping method for rest of the world at `wp-admin/admin.php?page=wc-settings&tab=shipping&section` 
<img width="1572" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/59fc6d4c-f00d-47dd-9b91-47e57e6f76a7">

8. Visit the site in a private window as a guest user.
9. Add the product to the cart and visit the Cart page.
10. Confirm it displays the shipping calculator with the `Calculate shipping for your location` link.

<img width="1518" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/5e41542a-f5dc-41b2-a3cf-c1ed20a38855">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Remove Active Shipping Zones check for displaying shipping calculator on the Cart Page. 
</details>

<details>

<summary>Changelog Entry Comment</summary>
Remove Active Shipping Zones check for displaying shipping calculator on the Cart Page. 
#### Comment 

</details>
